applications/tari_base_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -402,25 +402,59 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
             for (start, end) in page_iter {
                 debug!(target: LOG_TARGET, "Page: {}-{}", start, end);
                 // TODO: Better error handling
-                let result_headers = match handler.get_headers(start..=end).await {
+                let result_data = match handler.get_blocks(start..=end).await {
                     Err(err) => {
                         warn!(target: LOG_TARGET, "Internal base node service error: {}", err);
                         return;
                     },
                     Ok(data) => {
                         if is_reversed {
-                            data.into_iter().rev().collect::<Vec<_>>()
+                            data.into_iter()
```

Comment:
Something like this should DRY up the code
```suggestion
                            let mut data_iter = data.into_iter();
                            if is_reversed {
                            data_iter = data_iter.rev();
                            }
                            data_iter.map(|chain_block| { // etc.
```

Reply:
Oh that wont work - this is fine for now